cmd/commands/runtime.go

```diff
@@ -320,193 +352,213 @@ func createRBAC(repofs fs.FS, path, runtimeName string) error {
+			},
+			"workflow": {
+				Group:     "argoproj.io",
+				Version:   "v1alpha1",
+				Resource:  "workflows",
+				Namespace: runtimeName,
+				Selectors: selectors,
+			},
+			"appProject": {
+				Group:     "argoproj.io",
+				Version:   "v1alpha1",
+				Resource:  "appprojects",
+				Namespace: runtimeName,
+				Selectors: selectors,
+			},
+			"application": {
+				Group:     "argoproj.io",
+				Version:   "v1alpha1",
+				Resource:  "applications",
+				Namespace: runtimeName,
+				Selectors: selectors,
+			},
+			"eventBus": {
+				Group:     "argoproj.io",
+				Version:   "v1alpha1",
+				Resource:  "eventbus",
+				Namespace: runtimeName,
+				Selectors: selectors,
+			},
+			"eventSource": {
+				Group:     "argoproj.io",
+				Version:   "v1alpha1",
+				Resource:  "eventsources",
+				Namespace: runtimeName,
+				Selectors: selectors,
+			},
+			"sensor": {
+				Group:     "argoproj.io",
+				Version:   "v1alpha1",
+				Resource:  "sensors",
+				Namespace: runtimeName,
+				Selectors: selectors,
 			},
 		},
+	})
+	if err := insFs.WriteYamls(insFs.Join(resPath, "event-source.yaml"), eventSource); err != nil {
+		return err
 	}
 
-	return repofs.WriteYamls(repofs.Join(path, "sensor.yaml"), sensor)
+	sensor := eventUtils.CreateSensor(&eventUtils.CreateSensorOptions{
+		Name:            gsName + "-sensor",
+		Namespace:       runtimeName,
+		EventSourceName: eventSourceName,
+		EventBusName:    store.Get().EventBusName,
+		TriggerURL:      cfConfig.GetCurrentContext().URL + store.Get().EventReportingEndpoint,
+		Triggers: []string{
+			"clusterWorkflowTemplate",
+			"workflowTemplate",
+			"workflow",
+			"appProject",
+			"application",
+			"eventBus",
+			"eventSource",
+			"sensor",
+		},
+	})
+	if err = insFs.WriteYamls(insFs.Join(resPath, "sensor.yaml"), sensor); err != nil {
+		return err
+	}
+
+	gsPath := gsCloneOpts.FS.Join(apstore.Default.AppsDir, gsName, runtimeName)
+	fullGsPath := gsCloneOpts.FS.Join(gsCloneOpts.FS.Root(), gsPath)[1:]
+	app := cdUtils.CreateApp(&cdUtils.CreateAppOptions{
+		Name:      gsSyncName,
+		Namespace: runtimeName,
+		RepoURL:   gsCloneOpts.URL(),
+		Revision:  gsCloneOpts.Revision(),
+		SrcPath:   fullGsPath,
+	})
+	if err = insFs.WriteYamls(insFs.Join(resPath, gsName+"-synchronize.yaml"), app); err != nil {
+		return err
+	}
+
+	if err = billyUtils.WriteFile(gsFs, gsFs.Join(gsPath, "DUMMY"), []byte{}, 0666); err != nil {
+		return err
+	}
+
+	err = insRepo.Persist(ctx, &git.PushOptions{
+		CommitMsg: fmt.Sprintf("Created %s Resources", gsName),
+	})
+	if err != nil {
+		return err
+	}
+
+	err = gsRepo.Persist(ctx, &git.PushOptions{
+		CommitMsg: fmt.Sprintf("Created %s Directory", gsPath),
+	})
+	if err != nil {
+		return err
+	}
+
+	fullResPath := insFs.Join(insFs.Root(), resPath)
+	if err = createApp(ctx, insCloneOpts, runtimeName, gsName, insCloneOpts.URL()+fullResPath, application.AppTypeDirectory, runtimeName); err != nil {
+		return fmt.Errorf("failed to create git-source: %w", err)
```

Comment:
What happens if some of the steps failed?

Reply:
Currently, each autopilot command (bootstrap, project create, 3 * app create + 1 final app create for the git-source) performs its own clone-persist-commit-push. So a failure in a later step will still leave the GitOps repo with some data in it, and the cluster with some state of the runtime.
We thought about using a single FS for all the commands, and change the autopilot so it will run in a mode that just uses an existing local git repo, with pull-persist-commit only, and then just manually push at the end. This way the GitOps repo will either contain all the manifests, or none.